sycl/test-e2e/KernelCompiler/opencl_queries.cpp

```diff
@@ -0,0 +1,47 @@
+//==------- ocloc_queries.cpp ----------------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: ocloc
+
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// Note: the ocloc queries that are tested here are relatively new ( Dec 2023 )
+// if encountering many failures then an outdated ocloc is the likely culprit.
+
+#include <sycl/sycl.hpp>
+namespace syclex = sycl::ext::oneapi::experimental;
+
+int main() {
+  sycl::queue q;
+  sycl::device d = q.get_device();
+
+  assert(d.ext_oneapi_can_compile(syclex::source_language::opencl) &&
+         "can_compile(opencl) unexpectedly false");
+
+  assert(d.ext_oneapi_supports_cl_c_version(syclex::opencl_c_1_0) &&
+         "supports_cl_version(1.0) unexpectedly false");
+
+  assert(d.ext_oneapi_supports_cl_c_feature("__opencl_c_int64") &&
+         "int64 support unexpectedly false");
+  assert(!d.ext_oneapi_supports_cl_c_feature("not_a_real_feature") &&
+         "imaginery feature support unexpectedly true");
+
+  syclex::cl_version version{20, 20, 20};
+  bool res = d.ext_oneapi_supports_cl_extension("cl_intel_bfloat16_conversions",
+                                                &version);
+  if (res) {
+    assert(version.major != 20 && version.minor != 20 && version.patch != 20 &&
+           "version not updated");
+  }
+
+  // no supported devices support EMBEDDED_PROFILE at this time.
+  assert(d.ext_oneapi_cl_profile().find("FULL_PROFILE") == 0);
```

Comment:
```suggestion
  assert(d.ext_oneapi_cl_profile() == "FULL_PROFILE");
```

The `ext_oneapi_cl_profile` query returns just a single profile, not a list of them.  Therefore, I think you should test for equality.

Reply:
It returns FULL_PROFILE\n\n  at the moment.

Reply:
This seems like a bug someplace.  Does the OpenCL query `CL_DEVICE_PROFILE` return a string with the trailing newlines?  If so, it may be a bug in OpenCL.  If not, the bug may be in your code or in the new `ocloc` queries.

Reply:
It comes from OpenCL query that way.  Even if you call it directly on the command line.

Reply:
I just ran a test using code like this:

```
char profile[1024];
size_t len;
ret = clGetDeviceInfo(device_id, CL_DEVICE_PROFILE, sizeof(profile), &profile[0], &len);
if (ret != CL_SUCCESS) {
  fprintf(stderr, "ERROR: clGetDeviceInfo = %d\n", ret);
  return -1;
}
```

The OpenCL driver returns the string "FULL_PROFILE" with no terminating newline characters, which is also what the OpenCL specification says.

I think there is a bug someplace in the implementation of `device::ext_oneapi_cl_profile`.  If the `ocloc` tool is returning the newlines, the bug might be in `oloc`, in which case you should report it to the driver team.

If we need a temporary fix until we get an updated `ocloc`, I think it would be acceptable for the SYCL runtime to filter out the newlines before returning the string from `device::ext_oneapi_cl_profile`.  However, we should not write the test to expect the buggy return value.

Reply:
I added the trimming and reported it